src/linkage.md

```diff
@@ -211,6 +211,21 @@ a statically linked binary on MSVC you would execute:
 RUSTFLAGS='-C target-feature=+crt-static' cargo build --target x86_64-pc-windows-msvc
 ```
 
+## Prohibited linkage scenarios
+
+No crate may be linked with [the `panic=abort` runtime][panic-runtime] if it has
+both of the following characteristics:
+
+* It contains a call to an `-unwind` foreign function or function pointer
+* It was compiled with `panic=unwind`
+
+Note: Cargo will automatically unify all crates to use the same `panic`
+runtime, so this prohibition does not apply to projects compiled with Cargo.
+
+<!-- TODO: edit the above to indicate that this prohibition is enforced by
+`rustc` once https://github.com/rust-lang/rust/pull/97235 has been merged -->
```

Comment:
This TODO should be removed. And perhaps, this is a good place to mention about the `ffi-unwind-calls` lint?